### DISCUSSION TITLE
Performance

### DISCUSSION THREAD

**Discussion by tschortschde** [Zefau/ioBroker.jarvis#2407] [Q&A]
Hallo,
seit ich folgendes Script "Json" verwende stelle ich fest, dass ich deutliche Performance-Einbußen habe.
Habe ich hier evtl. einen Fehler im Script bzw. ist das der Fall, wenn aufwendige "JSon" geschrieben werden?

![image](https://github.com/Zefau/ioBroker.jarvis/assets/11274233/533074ba-2e3c-46e5-9580-9efb62f75c5b)

**Comment by mcuiobroker** [comment]
Welcher DP ist denn im Datenpunkt(Anzeige)?

**Comment by tschortschde** [comment]
knx.0.001_Büro.3_Beschattung_Fenster.E01_Büro_Ost_-_Status_aktuelle_Position

**Comment by mcuiobroker** [comment]
Dann ersetze den mal durch -> {val}

**Comment by tschortschde** [comment]
> Dann ersetze den mal durch -> {val}

Dann verschwindet die Statusrückmeldung-Rückmeldung
Vor dem Ersetzen:
![image](https://github.com/Zefau/ioBroker.jarvis/assets/11274233/4495f4d0-db6d-4475-b08f-5b5529744337)
Nach dem Ersetzen mit
![image](https://github.com/Zefau/ioBroker.jarvis/assets/11274233/0e50ba76-e391-4b15-8ede-79c6c3980f2e):
![image](https://github.com/Zefau/ioBroker.jarvis/assets/11274233/6cc62fbe-eecc-48ce-9611-1f5c3a6b12f0)

**Reply by tschortschde** [reply]
Irgendeine Idee?

**Comment by mcuiobroker** [comment]
So doch nicht.
Im Icon 
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/88db0a1f-d16e-43e7-9f39-4092aff8fe77)
siehe Beispiel aus der anderen Diskussion.
Performance dann besser?

**Comment by tschortschde** [comment]
Performance ist grundsätzlich besser...
Der Aufruf im Backend funktioniert schneller... jedoch habe ich jetzt ein anderes Problem...

Habe es wie empfohlen angepasst:
![image](https://github.com/Zefau/ioBroker.jarvis/assets/11274233/faf43139-c385-416b-ad42-6b1322647ac5)
Datenpunkt (Anzeige):
knx.0.001_Büro.3_Beschattung_Fenster.E01_Büro_Ost_-_Status_aktuelle_Position
Datenpunkt (Trigger):
knx.0.001_Büro.3_Beschattung_Fenster.E01_Büro_Ost_-_absolut
Icon (JSON Format):
{
  "{val}>50 && {knx.0.001_Büro.3_Beschattung_Fenster.E01_Büro_Ost_-_Verfahrstatus}=true": "window-shutter blink",
  "{val}<=50 && {knx.0.001_Büro.3_Beschattung_Fenster.E01_Büro_Ost_-_Verfahrstatus}=true": "window-shutter-open blink",
  "{val}>50 && {knx.0.001_Büro.3_Beschattung_Fenster.E01_Büro_Ost_-_Verfahrstatus}=false": "window-shutter",
  "{val}<=50 && {knx.0.001_Büro.3_Beschattung_Fenster.E01_Büro_Ost_-_Verfahrstatus}=false": "window-shutter-open",
  "default": "window-shutter"
}

Wenn ich die Jalousie fahren lasse, blinkt das Icon erst nach einer kurzen Zeit, obwohl der Status sofort "in Bewegung anzeigt"
Beim Stoppen blinkt die Jalousie einfach weiter...
https://github.com/Zefau/ioBroker.jarvis/assets/11274233/7cd4a669-779c-4772-84ac-5ae3f544974c

Was mache ich nur falsch?

**Comment by mcuiobroker** [comment]
Der type vom DP `knx.0.001_Büro.3_Beschattung_Fenster.E01_Büro_Ost-Verfahrstatus` ist boolean?
Einmal den STOP-Geräte-DP als Bild .

**Comment by tschortschde** [comment]
Ja, DP ist boolean:
![image](https://github.com/Zefau/ioBroker.jarvis/assets/11274233/5a25dac4-0061-42a4-a8a4-dbcee2a99328)

STOP-Geräte-DP
![image](https://github.com/Zefau/ioBroker.jarvis/assets/11274233/76fd7871-fa06-4334-b991-d23909dd21cd)
Datenpunkt Trigger:
knx.0.001_Büro.0_Zentral.E01_Büro_Ost_-_Jalousie_-_zentrale_Funktion_-_Stopp

**Comment by mcuiobroker** [comment]
Muss da nicht der verfahrstatus rein?
Wenn die Jalousie fährt ist der Verfahrstatus true?
![image](https://github.com/Zefau/ioBroker.jarvis/assets/84818934/6b7e6006-3422-4e0a-a1cf-c7dc413168ea)
Lamellenposition wäre was für den level?

**Reply by tschortschde** [reply]
Habe ich gemacht. 
Jedoch hat sich das Verhalten 

> Wenn ich die Jalousie fahren lasse, blinkt das Icon erst nach einer kurzen Zeit, obwohl der Status sofort "in Bewegung anzeigt"
Beim Stoppen blinkt die Jalousie einfach weiter...
https://github.com/Zefau/ioBroker.jarvis/assets/11274233/7cd4a669-779c-4772-84ac-5ae3f544974c

nicht geändert !?

**Comment by mcuiobroker** [comment]
`knx.0.001_Büro.0_Zentral.E01_Büro_Ost_-Jalousie-zentrale_Funktion-_Stopp` dieser DP ist ein Button?
Welchen Status hat der Verfahr-DP, wenn die Jalousie fährt?
Welcher DP zeigt `in bewegung` an?

**Comment by tschortschde** [comment]
>`knx.0.001_Büro.0_Zentral.E01_Büro_Ost_-Jalousie-zentrale_Funktion-_Stopp` dieser DP ist ein Button?

Du meinst im iobroker?:
![image](https://github.com/Zefau/ioBroker.jarvis/assets/11274233/200e8359-4ed7-42d1-b435-a3a2e4b82cb3)

>Welchen Status hat der Verfahr-DP, wenn die Jalousie fährt?

true

>Welcher DP zeigt `in bewegung` an?

![image](https://github.com/Zefau/ioBroker.jarvis/assets/11274233/8d957afe-e382-4a49-abf0-bdd3ccb9fc0b)

**Comment by mcuiobroker** [comment]
Wie lange dauert es denn bis der DP Verfahrstatus auf false geht, nachdem du die zentraleFunktion_Stopp gedrückt hast in ioBroker nicht jarvis?

**Reply by tschortschde** [reply]
Sofort

**Comment by mcuiobroker** [comment]
Da ist im Layout keine Zuordnung, die hinterher geändert wurde im LevelBody oder BlindLevelAction?
Sonst den Layout-Eintrag neu anlegen?

**Reply by tschortschde** [reply]
Nein, aber sicherheitshalber neu angelegt...
Keine Verbesserung ;-(

**Comment by mcuiobroker** [comment]
Test?
Start über jarvis.
Stop über ioBroker.
Jarvis und ioBroker-Objekte nebeneinander,wenn möglich das als GIF.
Mit gehen die Ideen aus.

**Comment by mcuiobroker** [comment]
Hast du das mal gemacht?

**Comment by tschortschde** [comment]
Bin leider noch nicht dazu gekommen. 
Stelle aber fest, dass bei einem Restart der Instanz das Blinken beim Start der Jalousien sofort angeht und beim stoppen sofort ausgeht!
Da gehts.
Sobald jedoch die Instanz etwas läuft wird das Verhalten träger ;-)

**Comment by mcuiobroker** [comment]
Das ist dann was für @zefau.

**Reply by tschortschde** [reply]
Okay, muss das als Issue oder so in seinem Repository eingestellt werden?

**Comment by mcuiobroker** [comment]
Ja als Issue eintragen mit Hinweis auf die Discussion.
